pkg/mountutil/mountutil.go

```diff
@@ -49,90 +53,60 @@ type Processed struct {
 	Opts            []oci.SpecOpts
 }
 
-func ProcessFlagV(s string, volStore volumestore.VolumeStore, createDir bool) (*Processed, error) {
-	var (
-		res      Processed
-		src, dst string
-		options  []string
-	)
-
-	s = strings.TrimLeft(s, ":")
-	split := strings.Split(s, ":")
-	switch len(split) {
-	case 1:
-		dst = s
-		res.AnonymousVolume = idgen.GenerateID()
-		log.L.Debugf("creating anonymous volume %q, for %q", res.AnonymousVolume, s)
-		anonVol, err := volStore.Create(res.AnonymousVolume, []string{})
-		if err != nil {
-			return nil, fmt.Errorf("failed to create an anonymous volume %q: %w", res.AnonymousVolume, err)
-		}
-		src = anonVol.Mountpoint
-		res.Type = Volume
-	case 2, 3:
-		res.Type = Bind
-		src, dst = split[0], split[1]
-		if !strings.Contains(src, "/") {
-			// assume src is a volume name
-			res.Name = src
-			vol, err := volStore.Get(src, false)
-			if err != nil {
-				if errors.Is(err, errdefs.ErrNotFound) {
-					vol, err = volStore.Create(src, nil)
-					if err != nil {
-						return nil, fmt.Errorf("failed to create volume %q: %w", src, err)
-					}
-				} else {
-					return nil, fmt.Errorf("failed to get volume %q: %w", src, err)
-				}
-			}
-			// src is now full path
-			src = vol.Mountpoint
-			res.Type = Volume
-		}
-		if !filepath.IsAbs(src) {
-			log.L.Warnf("expected an absolute path, got a relative path %q (allowed for nerdctl, but disallowed for Docker, so unrecommended)", src)
-			var err error
-			src, err = filepath.Abs(src)
-			if err != nil {
-				return nil, fmt.Errorf("failed to get the absolute path of %q: %w", src, err)
-			}
-		}
-		if createDir {
-			if _, err := os.Stat(src); err != nil {
-				if !os.IsNotExist(err) {
-					return nil, fmt.Errorf("failed to stat %q: %w", src, err)
-				}
-				if err := os.MkdirAll(src, 0o755); err != nil {
-					return nil, fmt.Errorf("failed to mkdir %q: %w", src, err)
-				}
-			}
-		}
+type volumeSpec struct {
+	Type            string
+	Name            string
+	Source          string
+	AnonymousVolume string
+}
 
-		if !filepath.IsAbs(dst) {
-			return nil, fmt.Errorf("expected an absolute path, got %q", dst)
-		}
-		rawOpts := ""
-		if len(split) == 3 {
-			rawOpts = split[2]
-		}
-		res.Mode = rawOpts
+func ProcessFlagV(s string, volStore volumestore.VolumeStore, createDir bool) (*Processed, error) {
+	split, err := splitVolumeSpec(s)
+	if err != nil {
+		return nil, fmt.Errorf("failed to split volume mount specification %q: %v", s, err)
+	}
 
-		// always call parseVolumeOptions for bind mount to allow the parser to add some default options
-		var err error
-		var specOpts []oci.SpecOpts
-		options, specOpts, err = parseVolumeOptions(res.Type, src, rawOpts)
-		if err != nil {
-			return nil, fmt.Errorf("failed to parse volume options (%q, %q, %q): %w", res.Type, src, rawOpts, err)
-		}
-		res.Opts = append(res.Opts, specOpts...)
-	default:
+	if len(split) < 1 || len(split) > 3 {
 		return nil, fmt.Errorf("failed to parse %q", s)
```

Comment:
Please give this a more descriptive error message like:

`fmt.Errorf("volume arg %q has was split into an incorrect number of elements (expected 1-3, got %d): %v", s, len(split), split)`

Reply:
This method is used to parse --mount and --tmpfs flags which have different formats. The [nerdctl command reference](https://github.com/containerd/nerdctl/blob/main/docs/command-reference.md#whale-blue_square-nerdctl-run) clearly documents the format of each input